src/pageComponents/login/index.tsx

```diff
@@ -46,24 +45,38 @@ const LoginPage = () => {
 		register,
 		handleSubmit,
 		formState: { errors },
-		// reset,
-	} = useForm<{ email: string; password: string }>();
+	} = useForm<{ email: string; password?: string }>();
+
 	const onSubmit: SubmitHandler<{
 		email: string;
-		password: string;
+		password?: string;
 	}> = async (data) => {
 		setIsLoading(true);
-		const { error } = await signInWithEmailPassword(
-			data?.email,
-			data?.password,
-			supabase,
-		);
+		const { error } = await signInWithOtp(data?.email, supabase);
 		setIsLoading(false);
 
 		if (error) {
 			errorToast(error?.message);
 		} else {
-			void router?.push(`/${ALL_BOOKMARKS_URL}`);
+			setEmail(data.email);
+			setIsOtpStep(true);
+		}
+	};
+
+	const handleVerifyOtp = async () => {
+		if (!otp || !email) {
+			errorToast("Please enter OTP");
+			return;
+		}
+
+		setIsLoading(true);
+		const { error } = await verifyOtp(email, otp, supabase);
+		setIsLoading(false);
+
+		if (error) {
+			errorToast(error.message);
+		} else {
+			window.location.pathname = `/${ALL_BOOKMARKS_URL}`;
```

Comment:
Shouldn't we use the next `router.push('/about')`, so that it doesn't do a full page navigation after verification?